### DISCUSSION TITLE
Multi resource tracking

### DISCUSSION THREAD

**Discussion by deadcorpseeater** [azerothcore/azerothcore-wotlk#15485] [Ideas]
Previouslly I used this repack https://www.emucoach.com/forums/threads/azerothcore-wotlk-solofriendly-repack-reborn.8464/ and found out that there is a possibility to use both mineral and herb tracking in game. That would be awesome for solo game, cuz a lot of players use all 11 professions. So I guess is it module or lua script? and how can I realize that? I guess there should be just change of spell such as mineral/herb tracking, cuz when u use it there is an action to turn of the other on tracking. Is there any possibility to get this? I'm kinda very noobie in that all

**Comment by BrucePrintscreen** [comment]
Hello there,

The tracking of herbs and minerals are indeed in the spell database.
You can change the spells by using Stoneharry's Spelleditor

https://github.com/stoneharry/WoW-Spell-Editor

If you look up the spell 'find herbs' and go to the 'Effects -> Effects 2 -> Spell Effects' you can see how this works

The most import value is the 'Misc Value'. 2 is for herbs and 3 is for minerals.

So if you want to find minerals while having the 'Find Herbs' active to achieve what you want is go to the spell effects 2 tab and change the values exactly as in the spell effects 1 tab, except for the Misc Value which you want to have the value 3.

On top of that you can rename 'find herbs' to 'find resources' or something similar to nake it look nice.

Good luck!

**Reply by dewmguy** [reply]
In TC this is performed with a world config parameter, which does not require changes to be made to the dbc or patch files of the client. Will these changes you've outlined here require changes to the client?

**Reply by dewmguy** [reply]
Additionally, I was hoping there would be a way to hack this ability in using Eluna. Any thoughts on the potential for a lua script?

**Reply by HMP92TT** [reply]
I actually managed to solve tracking both resources by editing SpellInfo.cpp and specifically the part IsAuraExclusiveBySpecificWith and then adding line AllowTrackBothResources = 1 in the worldserver.conf. The suggested comment guiding towards spell editing effects of the Find Herbs/Minerals spells haven't worked for me. The edited part is in the file below (with included Config.h include).

[SpellInfo.txt](https://github.com/user-attachments/files/22781228/SpellInfo.txt)

**Comment by Tntdruid** [comment]
@HMP92TT very nice.

Where do i place the SpellInfo.cpp?

**Reply by dewmguy** [reply]
leave it where it is. or you can create a mod that doesn't require you to edit the source code.

**Reply by deadcorpseeater** [reply]
ok, left to know how to create such a module)

I tried to use spelleditor and it was fine, but I guess it uses data from mysql server and I have individualprogression module with vanilla spellcosts I don't want to use. I guess if I Import it I'll break original spell.dbc on my server (with wotlk costs)

**Comment by vvo-01** [comment]
If anyone needs it, I found a simpler (though not very elegant) solution to this problem.
1. You'll need WDBXEditor (https://github.com/WowDevTools/WDBXEditor)
2. Use it to open the file (Your Server)\Data\dbc\Spell.dbc
3. Select the appropriate definition (WoTLK in my case)
4. Find the desired spell to edit ("Find Herbs" or "Find Minerals", ID 2383 or 2580, respectively)
In the spell you want to edit (one or both):
5. Set the "Effect_2" field to the same value as "Effect_1" (value = 6).
6. Set the "Implicit TargetA_2" field to the same value as "Implicit TargetA_1" (value = 1).
7. Set the "EffectAura_2" field to the same value as "EffectAura_1" (value = 45).
8. Edit the "EffectMiscValue_2" field:
a) If the "EffectMiscValue_1" field is 2, then set the "EffectMiscValue_2" field to 3 (for the "Find Herbs" spell, ID 2383).
b) If the "EffectMiscValue_1" field is 3, then set the "EffectMiscValue_2" field to 2 (for the "Find Minerals" spell, ID 2580).
9. Save the Spell.dbc file.

Then clear the WoW client cache and log in to the game. When you select the spell you modified, it will search for two types of resources instead of one.

<img width="206" height="67" alt="image" src="https://github.com/user-attachments/assets/b19d6cd8-be58-479e-a49b-3baca538c16a" />
<img width="244" height="63" alt="image" src="https://github.com/user-attachments/assets/ac79ffc8-3fd5-4195-9d34-b24bd271f61f" />
<img width="207" height="63" alt="image" src="https://github.com/user-attachments/assets/54480ee2-b918-4f09-afbe-27301bd9774f" />
<img width="250" height="65" alt="image" src="https://github.com/user-attachments/assets/af7dacd0-c0cb-49e9-a621-00766f4da4ab" />

**Comment by b-wun** [comment]
If anyone is still looking for a solution to this, I made a module: https://github.com/b-wun/mod-dual-tracking/tree/main

No DB or Core changes required, just clone it and build like you would any other Azeroth Core module :)
